skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+        using the query functionality. These locations are then dropped.
+
+        Parameters
+        ----------
+        locations : iterable of tuple of ints
+            A list of locations associated with the Interval object.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+        """
+        to_delete = {id(f) for f in
+                     self.query(locations=locations,
+                                metadata=metadata)}
+
+        new_intvls = []
+        # iterate through queries and drop them
+        for intvl in self._intervals:
+            if id(intvl) not in to_delete:
+                new_intvls.append(intvl)
+
+        self._intervals = new_intvls
+        self._is_stale_tree = True
+
+    @experimental(as_of='0.5.0-dev')
+    def __eq__(self, other):
+        '''Test if this object is equal to another.
+
+        This is performed by check if all the interval features
+        are equal between the two objects after sorting them by
+        locations.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for equality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are equal.
+        '''
+        self_intervals = sorted(self._intervals,
+                                key=operator.attrgetter('locations'))
+        other_intervals = sorted(other._intervals,
+                                 key=operator.attrgetter('locations'))
+        return self_intervals == other_intervals
+
+    @experimental(as_of='0.5.0-dev')
+    def __ne__(self, other):
+        '''Test if this object is not equal to another.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for inequality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are not equal.
+
+        See Also
+        --------
+        skbio.metadata.IntervalMetadata.__eq__
+        '''
+        return not self.__eq__(other)
+
+    @experimental(as_of='0.5.0-dev')
+    def __repr__(self):
+        '''Return a string representation of this object.
+
+        Returns
+        -------
+        str
+            String representation of this ``IntervalMetadata`` object.
+        '''
+        n = len(self._intervals)
+        l1 = '{} interval features'.format(n)
+        l2 = '-' * len(l1)
+
+        if n <= 5:
+            items = [repr(i) for i in self._intervals]
+        else:
+            items = [repr(self._intervals[i]) for i in [0, 1, 2, n-2, n-1]]
+            items[2] = '...'
+
+        return '\n'.join([l1, l2] + items)
+
+
+def _assert_valid_location(location):
+    if isinstance(location, tuple):
+        try:
+            start, end = location
+            if start > end:
+                raise ValueError("`start` is greater than `end`.")
+        except:
```

Comment:
Blanket exception catching can be dangerous because it masks certain errors like `KeyboardInterruptError`. Instead of a try/except, can you explicitly check the length of `location`?